File: Oct3/Example35/app/src/main/java/com/varunshankar/example35/WeatherTable.java
package com.varunshankar.example35;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

@Entity(tableName = "weather_table")
public class WeatherTable {
    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "location")
    private String location;

    @NonNull
    @ColumnInfo(name = "weatherdata")
    private String weatherJson;

    public WeatherTable(@NonNull String location, @NonNull String weatherJson){
        this.location = location;
        this.weatherJson = weatherJson;
    }

    public void setLocation(String location){
        this.location = location;
    }

    public void setWeatherJson(String weatherdata){
        this.weatherJson = weatherdata;
    }

    public String getLocation(){
        return location;
    }

    public String getWeatherJson(){
        return weatherJson;
    }
}

File: Oct1/Example35/app/src/main/java/com/example/example35/WeatherTable.kt
package com.example.example35

import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import androidx.room.Entity

// This does not need to be its own class. You could directly annotate the WeatherData class
// But since our Weather class is messy, it makes sense to strip out the parts
// that we want stored in the db (just for testing and maintainability).
// This does introduce the issue of drift between this class and the WeatherData class
@Entity(tableName = "weather_table")
data class WeatherTable(
    @field:ColumnInfo(name = "location")
    @field:PrimaryKey
    var location: String,
    @field:ColumnInfo(
    name = "weatherdata"
    ) var weatherJson: String
)